bin/dada2_denoising.R

```diff
@@ -0,0 +1,136 @@
+  
+  # Remove everything past '_trimmed'
+  sample_name <- strsplit(no_locus_filename, "_trimmed")[[1]][1]
+  
+  return(sample_name)
+}
+
+merge_with_concatenation <- function(dadaFs, derepFs, dadaRs, derepRs, amplicon_info) {
+  # Perform filtering
+  overlap_condition <- amplicon_info %>% filter((ampInsert_length + 10) < sum.mean.length.reads)
+  no_overlap_condition <- amplicon_info %>% filter((ampInsert_length + 10) >= sum.mean.length.reads)
+  
+  # Merge for overlap and no overlap conditions
+  mergers.overlap <- perform_merging(filter_samples(dadaFs, overlap_condition),
+                                     filter_samples(derepFs, overlap_condition),
+                                     filter_samples(dadaRs, overlap_condition),
+                                     filter_samples(derepRs, overlap_condition),
+                                     justConcatenate=FALSE)
+
+  mergers.no.overlap <- perform_merging(filter_samples(dadaFs, no_overlap_condition),
+                                        filter_samples(derepFs, no_overlap_condition),
+                                        filter_samples(dadaRs, no_overlap_condition),
+                                        filter_samples(derepRs, no_overlap_condition),
+                                        justConcatenate=TRUE)
+
+  # Combine the results
+  c(mergers.overlap, mergers.no.overlap)[names(dadaFs)]
+}
+
+perform_merging <- function(dadaFs_selected, derepFs_selected, dadaRs_selected, derepRs_selected, justConcatenate) {
+  mergePairs(dadaFs_selected, derepFs_selected, dadaRs_selected, derepRs_selected, 
+             verbose = TRUE, 
+             justConcatenate = justConcatenate, 
+             trimOverhang = TRUE,
+             minOverlap = 10,
+             maxMismatch = 1)
+}
+
+amplicon_data <- read.table(args$ampliconFILE, header=T)
+loci_list <- amplicon_data$amplicon
+
+if (args$just_concatenate) {
+  amplicon.info = data.frame(
+    names = sapply(strsplit(names(dadaFs),"_trimmed"),"[",1)
+  )
+
+  # Create amplicon info
+  amplicon.info <- amplicon.info %>%
+    mutate(names=sapply(str_split(names,'_S(\\d+)'),head,1)) %>% 
+    mutate(amplicon=unlist(lapply(str_split(names,'_'), function(x) { paste(x[1:3], collapse = "_") }))) %>%
+    inner_join(amplicon_data %>%
+               select(amplicon, ampInsert_length), by = c("amplicon")) %>%
+    select(amplicon, ampInsert_length) %>%
+    mutate(
+      sum.mean.length.reads = sapply(sapply(unlist(lapply(dadaFs,"[","sequence"),recursive = F),nchar),mean)+
+        sapply(sapply(unlist(lapply(dadaRs,"[","sequence"),recursive = F),nchar),mean)
+    )
+
+  rownames(amplicon.info) = names(dadaFs)
+
+  mergers <- merge_with_concatenation(dadaFs, derepFs, dadaRs, derepRs, amplicon.info)
+  
+  sample_name <- get_sample_name(names(derepFs)[1], loci_list)  # extract sample name
+  saveRDS(mergers, file = paste0(sample_name, "_merged.RDS"))
```

Comment:
I think we should add an output dir option to the script

Reply:
- [ ] add dout